### DISCUSSION TITLE
Zod is not able to catch decimal point error when it's optional()

### DISCUSSION THREAD

**Discussion by umerjaved178** [colinhacks/zod#3184] [Q&A] [answered]
I have 2 numeric fields, one is required and the other is optional

First Field: `z.number()`
Second Field: `z.number().optional()`

The problem comes while writing decimal numbers ( e.g: ".9", ".764"). On writing just ".", the first field throws an error but the second field doesn't!

I tried refining it and super refining it, transforming string to number etc but "." is not catchable when optional, nullable, bullish methods are used

Label: Bug Report

**Comment by JacobWeisenburger** [comment, marked as answer]
Please send a full reproducible code example.
